### ISSUE TITLE
Upload Error: urlopen error EOF occurred in violation of protocol

### ISSUE THREAD

**Issue by hash3liZer** [asciinema/asciinema#559]
I am trying to upload a cast file to the server but the uploading isn't working. I've checked across `MacOS` and `Debian` both has the same issue. Also, tried installing asciinema with `pip` and directly with `apt`. And tried versions `2.2.0` and `2.1.0` Nothing worked, same issue.  

This happens with a specific recording that i do. I've tried a basic file with simple `ls` command and that uploads just fine. Please note that i've changed nothing in the original file that is being created. Just trying to upload as it is. 

The recording that i am trying to do is a simple `subrake -d domain.com` command from [Subrake](https://github.com/hash3liZer/Subrake)

## To Reproduce
Installed the tool with pip, run auth command to login and upload the file: 
```bash
$ pip3 install asciinema
$ asciinema auth

# Starting recording and ran the tool
$ asciinema rec /tmp/file.cast
$ subrake -d zonetransfer.me
$ exit

# Uploading
$ asciinema upload castfile.cast
```

And i get the error:
<img width="962" alt="image" src="https://github.com/asciinema/asciinema/assets/29171692/ce33afe2-324c-48df-b08f-9924d36a2391">

**Versions:**
 - OS: [macOS Ventura 13.3, Kali 2023.1]
 - asciinema cli: [2.2.0, 2.2.1]
 - Python: 3.11

**Comment by sickill** [comment]
You're getting the exact same error on both macOS 13.3 and Kali 2023.1?

**Comment by sickill** [comment]
Can you attach a recording file which reproduces this error?

**Comment by hash3liZer** [comment]
Sure, this is the file that is getting failed. If i run with `asciinema play`, its plays fine and everything looks good. 

[file.zip](https://github.com/asciinema/asciinema/files/11927034/file.zip)

**Comment by sickill** [comment]
It's the size. It's 17MB, and the upload limit is 10MB currently. You should have got a nicer error message telling that but something have changed around server config which breaks that. I recently updated load balancer, I suspect this is when the actual message got lost.

Related issue https://github.com/asciinema/asciinema/issues/335

**Comment by sickill** [comment]
I've found why the actual "too big" error message is not printed. Python's urllib doesn't handle 413 response too well (it does, kinda, but can't cope with prematurely closed connection), and there seems to be no workaround we can make on the client side (this codebase). More on the problem here: https://github.com/psf/requests/issues/2422#issuecomment-184012765

I've made tweaks to server config though, which should give you more descriptive error message now.

Regarding the size limit: it's for several reasons, I explain it here in more detail: https://github.com/asciinema/asciinema/issues/91#issuecomment-335710575
